soh/soh/Enhancements/mods.cpp

```diff
@@ -486,6 +488,153 @@ void RegisterBonkDamage() {
+                GameInteractor::RawAction::FreezePlayer();
+                break;
+            case CHEST_BURN_TRAP:
+                GameInteractor::RawAction::BurnPlayer();
+                break;
+            case CHEST_SHOCK_TRAP:
+                GameInteractor::RawAction::ElectrocutePlayer();
+                break;
+            case CHEST_KNOCK_TRAP:
+                eventTimer = 3;
+                break;
+            case CHEST_SPEED_TRAP:
+                Audio_PlaySoundGeneral(NA_SE_VO_KZ_MOVE, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                GameInteractor::State::RunSpeedModifier = -2;
+                statusTimer = 200;
+                Overlay_DisplayText(10, "Speed Decreased!");
+                break;
+            case CHEST_BOMB_TRAP:
+                eventTimer = 3;
+                break;
+            case CHEST_VOID_TRAP:
+                Audio_PlaySoundGeneral(NA_SE_EN_GANON_LAUGH, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                eventTimer = 3;                    
+                break;
+            case CHEST_AMMO_TRAP:
+                eventTimer = 3;
+                Overlay_DisplayText(5, "Ammo Halved!");
+                break;
+            case CHEST_KILL_TRAP:
+                GameInteractor::RawAction::SetPlayerHealth(0);
+                break;
+            case CHEST_TELEPORT_TRAP:
+                eventTimer = 3;
+                break;
+        }
+    });
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnPlayerUpdate>([]() {
+        Player* player = GET_PLAYER(gPlayState);
+        if (statusTimer == 0) {
+            GameInteractor::State::RunSpeedModifier = 0;
+            statusTimer--;
+        } else {
+            statusTimer--;
+        }
+        if (eventTimer == 0) {
+            switch (roll) {
+                case CHEST_KNOCK_TRAP:
+                    GameInteractor::RawAction::KnockbackPlayer(1);
+                    break;
+                case CHEST_BOMB_TRAP:
+                    GameInteractor::RawAction::SpawnActor(ACTOR_EN_BOM, 1);
+                    break;
+                case CHEST_VOID_TRAP:
+                    Play_TriggerRespawn(gPlayState);
+                    break;
+                case CHEST_AMMO_TRAP:
+                    AMMO(ITEM_STICK) = AMMO(ITEM_STICK) * 0.5;
+                    AMMO(ITEM_NUT) = AMMO(ITEM_NUT) * 0.5;
+                    AMMO(ITEM_SLINGSHOT) = AMMO(ITEM_SLINGSHOT) * 0.5;
+                    AMMO(ITEM_BOW) = AMMO(ITEM_BOW) * 0.5;
+                    AMMO(ITEM_BOMB) = AMMO(ITEM_BOMB) * 0.5;
+                    AMMO(ITEM_BOMBCHU) = AMMO(ITEM_BOMBCHU) * 0.5;
+                    Audio_PlaySoundGeneral(NA_SE_VO_FR_SMILE_0, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                    break;
+                case CHEST_TELEPORT_TRAP:
+                    int entrance;
+                    int index = 1 + rand() % 10;
+                    if (index == 1) {
```

Comment:
This could be converted to a switch statement.  You can either have multiple cases all point to the same body like 

```
case 7:
case 8:
case 9:
``` 

or just have the Link's House option be the default and only make cases for options 1-6.

Reply:
Updated to a Switch statement, code looks a lot cleaner too.